### PR TITLE
rados/pool_tests: Verifiy Compression on EC pools and ratio set

### PR DESCRIPTION
**Pull request by pdhiran**
1. Verification of effects of compression on pool size
2. Verification of data size on pools with different "compression_required_ratio"

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass logs : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-K70RJ6/

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -943,6 +943,34 @@ def verify_ec_overwrites(self, **kwargs) -> bool:
...
+        Args:
+            pool_name: Name of the pool
+            **kwargs: additional params needed.
+                Allowed values:
+                    compression_required_ratio: ratio set on the pool for compression
+        Returns: True -> pass, False -> fail
+        """
+        log.info(f"Collecting stats about pool : {pool_name}")
+        pool_stats = self.run_ceph_command(cmd="ceph df detail")["pools"]
+        pool_1_stats = [detail for detail in pool_stats if detail["name"] == pool_name][
```
Comment (psathyan) [inline ceph/rados/core_workflows.py:959]: IMO, we should break as soon as we get a match. For example
```
for detail in pool_stats:
  if detail["name"] == pool_name:
    pool_1_stats = detail["stats"
    <you could include the below checks>
    break <return True -> if the below checks are included>
else:
  raise ResourceException("no matching detail found.")
```

File: tests/rados/pool_tests.py [hunk 2]
```diff
@@ -134,12 +134,24 @@ def run(ceph_cluster, **kw):
...
         pool_2 = pool_config["pool-2"]
-        if not rados_obj.create_pool(pool_name=pool_1, **pool_config):
-            log.error("could not create pool-1")
-            return 1
-        if not rados_obj.create_pool(pool_name=pool_2, **pool_config):
-            log.error("could not create pool-2")
-            return 1
+
+        if config["Compression_tests"]["pool_type"] == "replicated":
+            if not rados_obj.create_pool(pool_name=pool_1, **pool_config):
```
Comment (psathyan) [inline tests/rados/pool_tests.py:139]: Would it make sense to have a generic `create_pool` method and have it create `ec-pool` based on the argument provided to it else a regular osd pool.

Comment (pdhiran) [reply tests/rados/pool_tests.py:139]: Hi Satya, the ` create_pool ` method is already generic, and creates the pool, replicated or erasure, based on the args passed. What ` create_erasure_pool ` method does is, it creates a new EC profile for the pool to be created, verifies that the profile is created successfully, and then calls the same ` create_pool `  method, supplying it with the profile created.

Comment (psathyan) [reply tests/rados/pool_tests.py:139]: Thanks for details. It makes sense to have different methods.

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (pdhiran) [reply ceph/rados/core_workflows.py]: made the changes to break off from the loop once the pool is found.

**Review by psathyan — APPROVED** [review]

**Review by tintumathew10 — APPROVED** [review]

**Pull request merged**